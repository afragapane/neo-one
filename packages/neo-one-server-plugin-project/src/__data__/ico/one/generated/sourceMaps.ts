/* @hash 8a0e732eca2e3db44c8555ab42a98c67 */
// tslint:disable
/* eslint-disable */
/* @source-map-hash d5ed74ac9802fbf8b81fedd6c7bceb99 */
import { OneClient, SourceMaps } from '@neo-one/client';
import { projectID } from './projectID';

let sourceMapsIn: Promise<SourceMaps> = Promise.resolve({});
if (process.env.NODE_ENV !== 'production') {
  sourceMapsIn = Promise.resolve().then(async () => {
    const client = new OneClient(15624);
    const result = await client.request({
      plugin: '@neo-one/server-plugin-project',
      options: { type: 'sourceMaps', projectID },
    });

    return result.response;
  });
}

export const sourceMaps = sourceMapsIn;
