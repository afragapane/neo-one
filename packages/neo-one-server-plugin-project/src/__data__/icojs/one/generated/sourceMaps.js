/* @hash ee8a3e22d6a55c43b3a74efd2f406866 */
// tslint:disable
/* eslint-disable */
/* @source-map-hash 58e693f79dbf19e60ce908e01e160838 */
import { OneClient } from '@neo-one/client';
import { projectID } from './projectID';

let sourceMapsIn = Promise.resolve({});

if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
  sourceMapsIn = Promise.resolve().then(async () => {
    const client = new OneClient(10001);
    const result = await client.request({
      plugin: '@neo-one/server-plugin-project',
      options: { type: 'sourceMaps', projectID },
    });

    return result.response;
  });
}

export const sourceMaps = sourceMapsIn;
