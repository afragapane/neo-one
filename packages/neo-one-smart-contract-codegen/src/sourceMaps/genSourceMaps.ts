import { SourceMaps } from '@neo-one/client-common';

export const genSourceMaps = ({ sourceMaps }: { readonly sourceMaps: SourceMaps }) => ({
  js: `
let sourceMapsIn = {};

if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
  sourceMapsIn = ${JSON.stringify(sourceMaps)};
}

export const sourceMaps = sourceMapsIn;
`,
  ts: `
import { SourceMaps } from '@neo-one/client';

export const sourceMaps: SourceMaps;
`,
});
