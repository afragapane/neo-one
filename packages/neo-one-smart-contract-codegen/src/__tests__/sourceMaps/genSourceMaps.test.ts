import { genSourceMaps } from '../../sourceMaps';

describe('genSourceMaps', () => {
  test('Token', () => {
    expect(
      genSourceMaps({
        sourceMaps: {},
      }),
    ).toMatchSnapshot();
  });
});
