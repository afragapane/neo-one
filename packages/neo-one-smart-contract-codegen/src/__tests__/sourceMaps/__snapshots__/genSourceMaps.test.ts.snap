// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genSourceMaps Token 1`] = `
Object {
  "js": "
let sourceMapsIn = {};

if (process.env.NODE_ENV !== 'production' || process.env.NEO_ONE_DEV === 'true') {
  sourceMapsIn = {};
}

export const sourceMaps = sourceMapsIn;
",
  "ts": "
import { SourceMaps } from '@neo-one/client';

export const sourceMaps: SourceMaps;
",
}
`;
